pkg/components/beyla.go

```diff
@@ -114,9 +114,9 @@ func buildCommonContextInfo(
 			config.Attributes.Kubernetes.InformersSyncTimeout,
 		),
 	}
-	if config.InternalMetrics.Prometheus.Port != 0 {
+	if config.InternalMetrics.Prometheus.Port != 0 || config.Prometheus.Registry != nil {
```

Comment:
This would enable internal metrics by default when using Alloy. Any opinions against it?

Reply:
My only concern is open port for a container which is running with elevated permissions. It could increase the attack surface by default.